### DISCUSSION TITLE
Radius and massDensity parameters Question

### DISCUSSION THREAD

**Discussion by Junang-Wang** [sofa-framework/sofa#4421] [BeamAdapter] [answered]
Hi,
I am currently reviewing the BeamAdapter examples. In the "Tool_from_loader.scn" and "Single BeamDeployment.scn" examples, I noticed that the radius and massDensity settings in "RodStraightSection," "WireBeamInterpolation," and "AdaptiveBeamForceFieldAndMass" are different. This has caused some confusion for me. Do these parameters have different meanings, or is it just a typo?
![截图 2024-01-09 10-17-31](https://github.com/sofa-framework/sofa/assets/89964840/e90e5b6d-2389-4f24-ba3d-e92f2942796d)

**Comment by epernod** [comment, marked as answer]
Hi,
this is very good question. In fact I'm currently moving all the geometric and mechanical parameters inside the `RodxxxSection` components to be able to define those parameter per type of section. 

**Edit: I doubled checked:**
Right now for v23.06 and v23.12 the `radius ` and `massDensity ` used are the one in the `WireBeanInterpolation `and `AdaptiveBeamForceFieldAndMass ` components. 
The parameters in the scenes you are referring is a bad merge from me because I updated the scenes from my branch where I'm working on moving those parameters.

Sorry about the misleading and thank you for your feedback

**Reply by Junang-Wang** [reply]
Thanks for you reply, it helps a lot.

**Reply by Junang-Wang** [reply]
Hi, @epernod 
One more question, does `WireBeamInterpolation` supports sections with different radius? Should I import the radius as a list in  `WireBeamInterpolation` ? Same question for massDensity as well.
